src/NuGet.Clients/VisualStudio.Facade/NuGetUIThreadHelper.cs

```diff
@@ -33,28 +33,29 @@ public static JoinableTaskFactory JoinableTaskFactory
         /// Retrieve the CPS enabled JoinableTaskFactory for the current version of Visual Studio.
         /// This overrides the default VsTaskLibraryHelper.ServiceInstance JTF.
         /// </summary>
-        public static void SetJoinableTaskFactoryFromService(IComponentModel componentModel)
+        public static void SetJoinableTaskFactoryFromService(IProjectServiceAccessor projectServiceAccessor)
         {
-            if (componentModel == null)
+            if (projectServiceAccessor == null)
             {
-                throw new ArgumentNullException(nameof(componentModel));
+                throw new ArgumentNullException(nameof(projectServiceAccessor));
             }
 
-            _joinableTaskFactory = new Lazy<JoinableTaskFactory>(() =>
+            if (_joinableTaskFactory == null)
             {
-                var projectServiceAccessor = componentModel.GetService<IProjectServiceAccessor>();
-
+                _joinableTaskFactory = new Lazy<JoinableTaskFactory>(() =>
```

Comment:
`SetJoinableTaskFactoryFromService` will never be called simultaneously as in we won't be creating multiple NuGet projects at a same time so it will always be in sequence so how do you see there's still a race condition?

I don't see any possibility of interruption which is why i haven't guarded with `lock` region.